### ISSUE TITLE
Unable to check for NaN

### ISSUE THREAD

**Issue by paniq** [zdevito/terra#134]
I'm unable to use the typical check for NaN (not a number) floating point numbers because ``s ~= s`` checks appear to be optimized out.

see this reproduce:

```lua


local nan = tonumber("NaN")

terra isnanf (x : float) : bool
    return x ~= x
end

terra isnan (x : double) : bool
    return x ~= x
end

-- optimized to returning false :-(
isnanf:disas()

-- pure lua test passes
assert(nan ~= nan)

-- pass
assert(isnan(1.5) == false)
-- pass
assert(isnanf(1.5) == false)
-- fail
assert(isnan(nan) == true)
-- fail
assert(isnanf(nan) == true)
```

**Comment by zdevito** [comment]
Should be fixed by ac615b05a3f3e264f4822e75d329c55c64dee49b

**Comment by paniq** [comment]
yay, thanks
